addons/mail/static/src/core/common/navigable_list.xml

```diff
@@ -18,8 +18,10 @@
                         t-on-mouseenter="() => this.onOptionMouseEnter(option_index)"
                         t-on-click="(ev) => this.selectOption(ev, option_index)"
                     >
+                        <hr class="my-2" t-if="option.isSpecial and props.options.length > 1"/>
```

Comment:
This doesn't scale super well... What about setting an optional `group` entry on `option`, so that the navigable list groups the options as needed.

All options that have the same `group` are together. The value also determines precedence among groups.